### PR TITLE
[Macros] Verify '-load-plugin-library' precedes '-plugin-path'

### PR DESCRIPTION
**Pull request by rintaro**
The actual change was done in #65453.
Just adding a test case to prevent regressions.

rdar://108285056

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: test/Macros/macro_mixedpluginpath.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+//#-- Prepare the macro dylib plugin.
+// RUN: %host-build-swift \
+// RUN:   -swift-version 5 \
+// RUN:   -emit-library -o %t/plugins_local/%target-library-name(MacroDefinition) \
+// RUN:   -module-name MacroDefinition \
+// RUN:   %t/src/MacroDefinition.int.swift \
+// RUN:   -g -no-toolchain-stdlib-rpath
+
+//#-- Check '-load-plugin-library' precedes '-plugin-path'
```
Comment (bnbarham) [inline test/Macros/macro_mixedpluginpath.swift:26]: `takes precedence` would be clearer to me, but I'm not overly fussed.

**Review by bnbarham — APPROVED** [review]

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Pull request merged**